packages/discord.js/typings/index.d.ts

```diff
@@ -4314,14 +4323,14 @@ export class GuildStickerManager extends CachedManager<Snowflake, Sticker, Stick
 }
 
 export class GuildMemberRoleManager extends DataManager<Snowflake, Role, RoleResolvable> {
-  private constructor(member: GuildMember);
+  private constructor(member: GuildMember | PartialGuildMember);
   public get hoist(): Role | null;
   public get icon(): Role | null;
   public get color(): Role | null;
   public get highest(): Role;
   public get premiumSubscriberRole(): Role | null;
   public get botRole(): Role | null;
-  public member: GuildMember;
+  public member: GuildMember | PartialGuildMember;
```

Comment:
While technically correct this degrades UX. Probably better to add a `Partialize`d version of this manager overriding this property and overriding the `roles` getter on `PartialGuildMember` with that partial manager.